cl/search/views.py

```diff
@@ -53,7 +54,23 @@ def get_solr_result_objects(cd, facet):
     search_type = cd['type']
     if search_type == 'o':
         si = ExtraSolrInterface(settings.SOLR_OPINION_URL, mode='r')
-        results = si.query().add_extra(**build_main_query(cd, facet=facet))
+
+        # This is a `related:<pks>` prefix query?
+        related_prefix_match = re.search(r'(^|\s)(?P<pfx>related:(?P<pks>(([0-9]+)(,[0-9]+)*)))($|\s)', cd['q'])
```

Comment:
This regex is officially over my head! Also, we don't want to compile it every time somebody does a search query. Can we move it somewhere else, compile it once, and make it into a [verbose regex](https://linux.die.net/diveintopython/html/regular_expressions/verbose.html) so people like me can make sense of it?

Reply:
I moved this over to search_utils, but it should still be cleaned up, please.

Reply:
I've moved the regex to constants and made it verbose.